code/game/objects/structures/stool_bed_chair_nest/chairs.dm

```diff
@@ -18,6 +18,25 @@
 	..()
 	handle_layer()
 
+/obj/structure/bed/chair/lock_atom(var/atom/movable/AM)
+	. = ..()
+	update_icon()
+
+/obj/structure/bed/chair/unlock_atom(var/atom/movable/AM)
+	..()
```

Comment:
This is going to completely turbofuck vehicles and possibly others, make sure the return value is passed along